xdist/dsession.py

```diff
@@ -722,17 +722,20 @@ def __init__(self, config):
         self.tr = config.pluginmanager.getplugin("terminalreporter")
         self._status = {}
         self._lastlen = 0
+        self._isatty = self.tr.hasmarkup
```

Comment:
on first gnalce i think that instead of hasattr, we should use `getattr(self.tr, 'isatty', self.tr.hasmarkup)`